app/src/main/java/com/concordium/wallet/onboarding/data/OnboardingRepository.kt

```diff
@@ -0,0 +1,22 @@
+package com.concordium.wallet.onboarding.data
+
+import com.concordium.wallet.onboarding.data.datasource.OnboardingService
+import com.concordium.wallet.onboarding.data.datasource.SharedPreferencesDataSource
+import com.concordium.wallet.onboarding.ui.terms.model.toItem
+
+class OnboardingRepository(
+    private val sharedPreferences: SharedPreferencesDataSource,
+    private val onboardingService: OnboardingService
+) {
+
+    fun saveAcceptedTermsAndConditionsVersion(version: String) = runCatching {
+        sharedPreferences.termsAndConditionsVersionAccepted = version
+    }
+
+    fun getLocalAcceptedTermsAndConditionsVersion(): String =
+        sharedPreferences.termsAndConditionsVersionAccepted
```

Comment:
I removed `suspend` as Android Studio claimed that they were redundant. Please confirm.

Reply:
ok feel free to re-add it then

Reply:
for the time being, we can leave it like this, I'll add this when necessary.